transforms/csdl/preprocess_csdl.xsl

```diff
@@ -668,6 +668,26 @@
             </Annotation>
         </xsl:copy>
     </xsl:template>
+    
+    <!--Delta function for messages needs the changeType parameter-->
+    <xsl:template match="edm:Schema[@Namespace='microsoft.graph']/edm:Function[@Name='delta'][edm:Parameter[@Name='bindingparameter']][edm:Parameter[@Type='Collection(graph.message)']]">
+        <xsl:copy>
+            <xsl:apply-templates select="@* | node()"/>
+            <Annotation Term="Org.OData.Capabilities.V1.OperationRestrictions">
+            <Record>
+                <PropertyValue Property="CustomQueryOptions">
+                    <Collection>
+                        <Record>
+                            <PropertyValue Property="Name" String="changeType" />
+                            <PropertyValue Property="Description" String="A custom query option to filter the delta response based on the type of change. Supported values are created, updated or deleted." />
+                            <PropertyValue Property="Required" Bool="true" />
```

Comment:
@irvinesunday I believe the "required" property here should be false as its not mandatory to pass this query parameter.

Reply:
I must have misinterpreted this [statement](https://learn.microsoft.com/en-us/graph/api/message-delta?view=graph-rest-1.0&tabs=http#query-parameters):

![image](https://github.com/microsoftgraph/msgraph-metadata/assets/40403681/e78ae754-4524-40f9-b930-024bb29bde0a)

I will push a PR to resolve it.